### PR TITLE
Small refactorings and cosmetic changes

### PR DESCRIPTION
**Pull request by fw42**
A few more cosmetic changes... I will comment on each one inline on the diff.

### CODE REVIEW

File: progress.go [hunk 1]
```diff
@@ -79,12 +94,17 @@ func (p *Progress) Report(s Stat) {
 	cur := p.cur
 	p.curM.Unlock()
 
-	// update progress
-	if p.OnUpdate != nil {
-		p.fnM.Lock()
-		p.OnUpdate(cur, time.Since(p.start), false)
-		p.fnM.Unlock()
+	p.updateProgress(cur, false)
```
Comment (fw42) [inline progress.go]: moved this stuff into a new function since there was some duplication (the almost exact same code existed below again)

File: progress.go [hunk 2]
```diff
@@ -111,40 +126,23 @@ func (p *Progress) reporter() {
...
-	p.curM.Unlock()
-}
-
 // Done closes the progress report.
 func (p *Progress) Done() {
 	if p == nil || !p.running {
 		return
 	}
 
-	if p.running {
```
Comment (fw42) [inline progress.go]: this check was unnecessary since you already check for `!p.running` above and then `return`

File: restorer.go [hunk 3]
```diff
@@ -19,57 +19,31 @@ type Restorer struct {
 	Filter func(item string, dstpath string, node *Node) bool
 }
 
-// NewRestorer creates a restorer preloaded with the content from the snapshot snid.
-func NewRestorer(s *server.Server, snid backend.ID) (*Restorer, error) {
-	r := &Restorer{s: s}
+var abortOnAllErrors = func(str string, node *Node, err error) error { return err }
+
+// NewRestorer creates a restorer preloaded with the content from the snapshot id.
+func NewRestorer(s *server.Server, id backend.ID) (*Restorer, error) {
```
Comment (fw42) [inline restorer.go]: you call this `id` (not `snid`) everywhere else, so I renamed it

File: restorer.go [hunk 4]
```diff
@@ -19,57 +19,31 @@ type Restorer struct {
 	Filter func(item string, dstpath string, node *Node) bool
 }
 
-// NewRestorer creates a restorer preloaded with the content from the snapshot snid.
-func NewRestorer(s *server.Server, snid backend.ID) (*Restorer, error) {
-	r := &Restorer{s: s}
+var abortOnAllErrors = func(str string, node *Node, err error) error { return err }
```
Comment (fw42) [inline restorer.go]: I think having it in a variable like this is nicer than adding a comment (also there was a bug: since this was defined as a closure, it was referencing the `err` from the surrounding context, not the error that was being passed into the function (it wasn't even named))

File: restorer.go [hunk 5]
```diff
@@ -19,57 +19,31 @@ type Restorer struct {
...
 	}
 
-	// abort on all errors
-	r.Error = func(string, *Node, error) error { return err }
-
 	return r, nil
 }
 
-func (res *Restorer) to(dst string, dir string, treeID backend.ID) error {
+func (res *Restorer) restoreTo(dst string, dir string, treeID backend.ID) error {
```
Comment (fw42) [inline restorer.go]: This function was quite long. I broke out the core part (the loop body for non-directories) into it's own function (`restoreNodeTo`). I think this increases readability.

File: snapshot_test.go [hunk 6]
```diff
@@ -2,29 +2,17 @@ package restic_test
...
 	. "github.com/restic/restic/test"
 )
 
-func testSnapshot(t *testing.T, s *server.Server) {
-	var err error
-	sn, err := restic.NewSnapshot([]string{"/home/foobar"})
-	OK(t, err)
-	// sn.Tree, err = restic.Blob{ID: backend.ParseID("c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2")}
-	// ok(t, err)
-	sn.Time, err = time.Parse(time.RFC3339Nano, "2014-08-03T17:49:05.378595539+02:00")
```
Comment (fw42) [inline snapshot_test.go]: I don't know what this is... was this even testing anything? I removed it... also removed the commented lines here.. if it's important that this gets done, we should add a github issue

File: tree.go [hunk 7]
```diff
@@ -74,7 +73,7 @@ func (t *Tree) Insert(node *Node) error {
 	return nil
 }
 
-func (t Tree) find(name string) (int, *Node, error) {
+func (t Tree) binarySearch(name string) (int, *Node, error) {
```
Comment (fw42) [inline tree.go]: renamed this for clarity, since it took me a few minutes to realize why this is not super slow

File: tree.go [hunk 8]
```diff
@@ -60,9 +60,8 @@ func (t Tree) Equals(other *Tree) bool {
 }
 
 func (t *Tree) Insert(node *Node) error {
-	pos, _, err := t.find(node.Name)
+	pos, _, err := t.binarySearch(node.Name)
 	if err == nil {
-		// already present
```
Comment (fw42) [inline tree.go]: I think the name of the error already says what this is doing, no need to have the comment

File: tree_test.go [hunk 9]
```diff
@@ -21,8 +21,7 @@ var testFiles = []struct {
 	{"bar/bla/blubb", []byte("This is just a test!\n")},
 }
 
-// prepareDir creates a temporary directory and returns it.
```
Comment (fw42) [inline tree_test.go]: again, renaming the function is more concise than having a comment on a function with an ambiguous name, imho.

File: tree_test.go [hunk 10]
```diff
@@ -89,7 +88,7 @@ func TestNodeComparison(t *testing.T) {
 	n2 := *node
 	Assert(t, node.Equals(n2), "nodes aren't equal")
 
-	n2.Size -= 1
+	n2.Size--
```
Comment (fw42) [inline tree_test.go]: golint doesn't like `-= 1` and `+= 1`

File: walk.go [hunk 11]
```diff
@@ -30,15 +30,15 @@ func walkTree(s *server.Server, path string, treeID backend.ID, done chan struct
 		if node.Type == "dir" {
 			walkTree(s, p, node.Subtree, done, jobCh)
 		} else {
-			jobCh <- WalkTreeJob{Path: p, Node: node, Error: err}
```
Comment (fw42) [inline walk.go]: `err` is always `nil` at this point, so no need to explicitly pass it

File: walk_test.go [hunk 12]
```diff
@@ -30,11 +30,6 @@ func TestWalkTree(t *testing.T) {
 	// flush server, write all packs
 	OK(t, server.Flush())
 
-	// start benchmark
-	// t.ResetTimer()
-
-	// for i := 0; i < t.N; i++ {
```
Comment (fw42) [inline walk_test.go]: unused code

File: archiver.go [hunk 13]
```diff
@@ -37,24 +39,20 @@ type Archiver struct {
...
 		blobToken: make(chan struct{}, maxConcurrentBlobs),
 	}
 
-	// fill blob token
 	for i := 0; i < maxConcurrentBlobs; i++ {
 		arch.blobToken <- struct{}{}
 	}
 
-	// abort on all errors
-	arch.Error = func(string, os.FileInfo, error) error { return err }
```
Comment (fw42) [inline archiver.go]: FYI I don't think this would have aborted anything, it would have returned `nil` in every case, even on errors...

Comment (fd0) [inline archiver.go]: Oh, nice catch!

File: restorer.go (hunk 4 above)
Comment (fd0) [inline restorer.go]: Great catch btw!

File: cache.go [hunk 14]
```diff
@@ -90,14 +82,14 @@ func (c *Cache) Load(t backend.Type, subtype string, id backend.ID) (io.ReadClos
 	return os.Open(filename)
 }
 
-func (c *Cache) Purge(t backend.Type, subtype string, id backend.ID) error {
+func (c *Cache) purge(t backend.Type, subtype string, id backend.ID) error {
```
Comment (fd0) [inline cache.go]: Why did you change the visibility of `Purge()`?

File: archiver_test.go [hunk 15]
```diff
@@ -121,10 +121,10 @@ func archiveDirectory(b testing.TB) {
 	key := SetupKey(b, server, "geheim")
 	server.SetKey(key)
 
-	arch, err := restic.NewArchiver(server)
-	OK(b, err)
+	arch := restic.NewArchiver(server)
 
 	_, id, err := arch.Snapshot(nil, []string{*benchArchiveDirectory}, nil)
+	OK(b, err)
```
Comment (fd0) [inline archiver_test.go]: This is another good catch!

**Comment by fd0** [comment]
Looks good to me. I'll merge it with one exception: I'll dropped the visibility change for `Purge()`.

**Comment by fw42** [comment]
> I'll dropped the visibility change for Purge().

There was no reason for it to be public, it wasn't being called from anywhere (except other methods of the same type).
